cmd/minikube/cmd/config/config.go

```diff
@@ -0,0 +1,130 @@
+package config
+
+import (
+	"encoding/json"
+	"io"
+
+	"os"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	"k8s.io/minikube/pkg/minikube/constants"
+)
+
+type configFile interface {
+	io.ReadWriter
+}
+
+type setFn func(string, string) error
+type MinikubeConfig map[string]interface{}
+
+type Setting struct {
+	name        string
+	set         func(MinikubeConfig, string, string) error
+	validations []setFn
+	callbacks   []setFn
+}
+
+// These are all the settings that are configurable
+// and their validation and callback fn run on Set
+var settings []Setting = []Setting{
+	Setting{
+		name:        "vm-driver",
+		set:         SetString,
+		validations: []setFn{IsValidDriver},
+		callbacks:   []setFn{RequiresRestartMsg},
+	},
+	Setting{
+		name:        "cpus",
+		set:         SetInt,
+		validations: []setFn{IsPositive},
+		callbacks:   []setFn{RequiresRestartMsg},
+	},
+	Setting{
+		name:        "disk-size",
+		set:         SetString,
+		validations: []setFn{IsValidDiskSize},
+		callbacks:   []setFn{RequiresRestartMsg},
+	},
+	Setting{
+		name:        "host-only-cidr",
+		set:         SetString,
+		validations: []setFn{IsValidCIDR},
+	},
+	Setting{
+		name:        "memory",
+		set:         SetInt,
+		validations: []setFn{IsPositive},
+		callbacks:   []setFn{RequiresRestartMsg},
+	},
+	Setting{
+		name: "show-libmachine-logs",
+		set:  SetBool,
+	},
+	Setting{
+		name:        "log_dir",
+		set:         SetString,
+		validations: []setFn{IsValidPath},
+	},
+	Setting{
+		name: "kubernetes-version",
+		set:  SetString,
+	},
+}
+
+var ConfigCmd = &cobra.Command{
+	Use:   "config SUBCOMMAND [flags]",
+	Short: "Modify minikube config",
+	Long:  `Modify minikube config using subcommands like "minikube config set verbosity 100"`,
+	Run: func(cmd *cobra.Command, args []string) {
+		cmd.Help()
+	},
+}
+
+// Reads in the JSON minikube config
+func ReadConfig() map[string]interface{} {
+	f, err := os.Open(constants.ConfigFile)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return make(map[string]interface{})
+		}
+		glog.Errorf("Could not open file %s: %s", constants.ConfigFile, err)
+	}
+	var m map[string]interface{}
```

Comment:
Should this be the MinikubeConfig type alias you defined above? (same for line 89)